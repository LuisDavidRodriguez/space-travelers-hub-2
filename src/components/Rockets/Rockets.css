.rocket-image {
  width: 10%;
}
