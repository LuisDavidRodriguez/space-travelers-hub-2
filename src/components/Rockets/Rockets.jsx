import React, { Fragment, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getRockets } from '../../redux/rockets/rocketSlice';
import SingleRocket from './SingleRockets';

const Rockets = () => {
  const allRockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!allRockets.length) {
      dispatch(getRockets());
    }
  }, [allRockets.length]);

  return (
    <>
      <div>Rockets List</div>
      <div className="rockets-container">
        {
        allRockets.length ? <SingleRocket allrocekts={allRockets} /> : <div>No Rockets Found</div>
      }
      </div>
    </>
  );
};
export default Rockets;
