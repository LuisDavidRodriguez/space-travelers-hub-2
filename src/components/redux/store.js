import { configureStore, combineReducers } from '@reduxjs/toolkit';
import RocketReducer from './rockets/rocketSlice';

const reducer = combineReducers({
  rockets: RocketReducer,
});

const store = configureStore({ reducer });

export default store;
