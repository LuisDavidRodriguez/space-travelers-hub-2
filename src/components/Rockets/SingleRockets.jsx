import React from 'react';
import { useDispatch } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Table, Stack, Badge } from 'react-bootstrap';
import { reserveRocket, cancelRocket } from '../../redux/rockets/rocketSlice';
import './Rockets.css';

const SingleRocket = ({ allrocekts }) => {
  const dispatch = useDispatch();

  return (
    <div>
      <Stack className="p-3">
        <Table responsive striped bordered hover size="sm">
          <thead>
            <tr>
              <th>Image</th>
              <th>Rockets</th>
              <th>Description</th>
              <th>Status</th>
              <th> </th>
            </tr>
          </thead>

          <tbody>
            {
        allrocekts.map((rocket) => (

          <tr key={rocket.id}>
            <td><img className="rocket-image" src={rocket.flickrImages} alt={rocket.rocket_name} /></td>
            <td className="fw-bold">{rocket.name}</td>
            <td>{rocket.description}</td>
            <td className="align-middle">
              <Badge bg={rocket.reserved ? 'secondary' : ''}>
                {rocket.reserved ? <span className="rocket-unavailable">RESERVED</span> : ' ' }
              </Badge>
            </td>
            <td className="align-middle">
              {rocket.reserved === false ? (
                <button
                  className="btn-reserve-rocket"
                  onClick={() => dispatch(reserveRocket(rocket.id))}
                  id={rocket.id}
                  type="submit"
                >
                  RESERVE DRAGON
                </button>
              ) : (
                <button
                  className="btn-cancel-reservation"
                  onClick={() => dispatch(cancelRocket(rocket.id))}
                  id={rocket.id}
                  type="submit"
                >
                  CANCEL DRAGON
                </button>
              )}
            </td>
          </tr>
        ))
        }
          </tbody>

        </Table>
      </Stack>

    </div>
  );
};

SingleRocket.propTypes = {
  allrocekts: PropTypes.shape(
    PropTypes.oneOfType([
      PropTypes.string.isRequired,
      PropTypes.bool.isRequired,
    ]),
  ).isRequired,
};

export default SingleRocket;
