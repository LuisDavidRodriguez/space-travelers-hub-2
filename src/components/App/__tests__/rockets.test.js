import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Rockets from '../../Rockets/Rockets';
import store from '../../../redux/configStore';

describe('Rockets Component Rendering ', () => {
  it('Matches snapshot', () => {
    const Tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <Rockets />
        </Provider>
      </MemoryRouter>,
    ).toJSON();
    expect(Tree).toMatchSnapshot();
  });
});
