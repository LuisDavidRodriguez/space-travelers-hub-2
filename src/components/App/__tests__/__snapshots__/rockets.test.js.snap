// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets Component Rendering  Matches snapshot 1`] = `
Array [
  <div>
    Rockets List
  </div>,
  <div
    className="rockets-container"
  >
    <div>
      No Rockets Found
    </div>
  </div>,
]
`;
